Test/Resolve-WhiskeyDotnetSdkVersion.Tests.ps1

```diff
@@ -210,3 +245,176 @@ Describe 'Resolve-WhiskeyDotNetSdkVersion.when Microsoft moves the index again'
         ThenReturnedNothing
     }
 }
+
+Describe 'Resolve-WhiskeyDotNetSdkVersion.when RollForward is diabled and patch is found' {
+    It 'should use specified version' {
+        Init
+        GivenReleases '3.0'
+        GivenSDKVersions '3.0.000', '3.0.100' -ForRelease '3.0'
+        WhenResolvingSdkVersion -Version '3.0.000' -RollForward Disable
+        ThenResolvedVersion '3.0.000'
+    }
+}
+
+Describe 'Resolve-WhiskeyDotNetSdkVersion.when RollForward is disabled and patch is not found' {
+    It 'should fail' {
+        Init
+        GivenReleases '3.0'
+        GivenSDKVersions '3.0.100' -ForRelease '3.0'
+        WhenResolvingSdkVersion -Version '3.0.000' -RollForward Disable
+        ThenError -Message 'not be found'
+    }
+}
+
+Describe 'Resolve-WhiskeyDotNetSdkVersion.when RollForward is patch and patch is found' {
+    It 'should use specified version' {
+        Init
+        GivenReleases '3.0'
+        GivenSDKVersions '3.0.000', '3.0.100' -ForRelease '3.0'

```

Comment:
You need multiple eligible versions here to ensure it's choosing the correct one, which should be the latest patch.

```powershell
GivenSDKVersions '3.0.000', '3.0.001', '3.0.100' -ForRelease '3.0'
WhenResolvingSdkVersion -Version '3.0.000' -RollForward Patch
ThenResolvedVersion '3.0.001'
```